src/compiler/checker.ts

```diff
@@ -7433,7 +7433,7 @@ namespace ts {
                     skippedPrivateMembers.set(rightProp.name, true);
                 }
                 else if (!isClassMethod(rightProp) && !isSetterWithoutGetter) {
-                    members.set(rightProp.name, rightProp);
+                    members.set(rightProp.name, getNonReadonlySymbol(rightProp));

```

Comment:
Or try to be smarter about creating spread types; you don't really need to create a new spread object type if the right side doesn't modify the types of the left side in any way, you can just return the left type.